src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -31,17 +33,49 @@ var AppModalComponent = React.createClass({
     rollBackApp: React.PropTypes.func.isRequired,
     scaleApp: React.PropTypes.func.isRequired,
     suspendApp: React.PropTypes.func.isRequired,
-    tasksFetchState: React.PropTypes.number.isRequired
+    tasksFetchState: React.PropTypes.number.isRequired,
+    router: React.PropTypes.object.isRequired
   },
 
   getInitialState: function () {
+    var appid = this.props.model.get("id");
+    var activeTabId;
+
+    var tabs = _.reduce(tabsTemplate, function (current, tab) {
```

Comment:
Again, shouldn't this be getting handled at a higher level? The component shouldn't be doing raw munging of the whole URL. It should just have a switch as to whether something is being shown or not.